### PR TITLE
Properly implement termination grace period seconds

### PR DESCRIPTION
**Pull request by eakmanrq**
For services except Redis, this PR results in no behavior change but allows users to configure `terminationGracePeriodSeconds` if they want.

Redis is unusual in that the chart had an entry for `terminationGracePeriodSeconds` already with a default of 600. So it seems the author wanted Redis to have a grace period of 600 but actually was getting 30 since it wasn't properly configured. Therefore I properly added the `terminationGracePeriodSeconds` for Redis which will change the behavior from 30 seconds (since one wasn't provided) to 600. I could also change the default to 30 to maintain current behavior but my interpretation of the chart is that current behavior is a bug. A longer termination grace period seconds for Redis can make sense if you want to ensure you don't have data loss.

---
**^ Add meaningful description above**
Read the **[Pull Request Guidelines](https://github.com/apache/airflow/blob/main/contributing-docs/05_pull_requests.rst#pull-request-guidelines)** for more information.
In case of fundamental code changes, an Airflow Improvement Proposal ([AIP](https://cwiki.apache.org/confluence/display/AIRFLOW/Airflow+Improvement+Proposals)) is needed.
In case of a new dependency, check compliance with the [ASF 3rd Party License Policy](https://www.apache.org/legal/resolved.html#category-x).
In case of backwards incompatible changes please leave a note in a newsfragment file, named `{pr_number}.significant.rst` or `{issue_number}.significant.rst`, in [newsfragments](https://github.com/apache/airflow/tree/main/newsfragments).

### CODE REVIEW

File: chart/values.schema.json [hunk 1]
```diff
@@ -2404,6 +2404,11 @@
                     ],
                     "default": null
                 },
+                "terminationGracePeriodSeconds": {
+                    "description": "Grace period for tasks to finish after SIGTERM is sent from Kubernetes.",
```
Comment (romsharon98) [inline chart/values.schema.json]: isn't is should be
```suggestion
                    "description": "Grace period for scheduler to finish after SIGTERM is sent from Kubernetes.",
```

File: chart/values.schema.json [hunk 2]
```diff
@@ -5363,6 +5368,11 @@
                     ],
                     "default": null
                 },
+                "terminationGracePeriodSeconds": {
+                    "description": "Grace period for webserver to finish after SIGTERM is sent from Kubernetes.",
+                    "type": "integer",
+                    "default": 30
```
Comment (romsharon98) [inline chart/values.schema.json:5374]: in all current deployment this value is 60.
why you choose 30? (apply for others)

**Comment by romsharon98** [comment]
can you please add tests for your changes?
you can find more information about testing helm [here](https://github.com/romsharon98/airflow/blob/main/contributing-docs/testing/helm_unit_tests.rst)

File: chart/values.schema.json (hunk 1 above)
Comment (eakmanrq) [reply chart/values.schema.json]: Thank you!

File: chart/values.schema.json (hunk 2 above)
Comment (eakmanrq) [reply chart/values.schema.json:5374]: Hmm I need to better understand why you are thinking the current value is 60. If you don't provide a `terminationGracePeriodSeconds` then kubernetes itself will default to 30.

**Comment by eakmanrq** [comment]
> can you please add tests for your changes? you can find more information about testing helm [here](https://github.com/romsharon98/airflow/blob/main/contributing-docs/testing/helm_unit_tests.rst)

Ah this is great, I missed there were tests for Helm. Will update with tests.

**Comment by eakmanrq** [comment]
@romsharon98 

Added tests and addressed typo: https://github.com/apache/airflow/pull/41374/commits/f980282c461a0f6424acc056930780388e426ea3

Looking for more feedback on the 60 second concern.

**Review by jedcunningham — APPROVED** [review]

File: chart/values.schema.json (hunk 2 above)
Comment (romsharon98) [reply chart/values.schema.json:5374]: I meant that the terminationGracePeriodSeconds in other components equal to 60. 
[tigerrer](https://github.com/apache/airflow/blob/main/chart/values.yaml#L1579)
[dag processor](https://github.com/apache/airflow/blob/main/chart/values.yaml#L1766)
It's better to keep `terminationGracePeriodSeconds` consistent across all components unless there's a stronger reason to do otherwise.

Comment (eakmanrq) [reply chart/values.schema.json:5374]: The reason why I didn't do this is because this would result in a behavior change for these components. My goal was to make `terminationGracePeriodSeconds` configurable for more components while preserving existing behavior. I'm open though if we would also like to change the existing behavior to make it more consistent if we want to do that too.

Comment (romsharon98) [reply chart/values.schema.json:5374]: Gotcha, let's keep it like you did.

**Review by romsharon98 — APPROVED** [review]
Great work!

File: airflow/providers/fab/auth_manager/cli_commands/user_command.py [hunk 3]
```diff
@@ -212,10 +212,10 @@ def users_import(args):
 
     users_created, users_updated = _import_users(users_list)
     if users_created:
-        print("Created the following users:\n\t{}".format("\n\t".join(users_created)))
+        print(f"Created the following users:\n\t{'\\n\\t'.join(users_created)}")
```
Comment (jedcunningham) [inline airflow/providers/fab/auth_manager/cli_commands/user_command.py]: Can you undo these changes @eakmanrq ? 1) its breaking the image build and 2) we try to keep PRs isolated if possible (this is a provider, while the rest is the helm chart).

Comment (eakmanrq) [reply airflow/providers/fab/auth_manager/cli_commands/user_command.py]: Yeah I actually didn't make those changes. I think someone else updated my branch. Either way I will reverse it.

Comment (eakmanrq) [reply airflow/providers/fab/auth_manager/cli_commands/user_command.py]: Oh I see running style did this. Must be a misconfiguration on my end. Will fix.

Comment (eakmanrq) [reply airflow/providers/fab/auth_manager/cli_commands/user_command.py]: Fixed: https://github.com/apache/airflow/pull/41374/commits/ac2af646d7866446e26498a7a01d0bf58e814c32

Thanks for catching that.

**Review by jedcunningham — APPROVED** [review]

**Comment by jedcunningham** [comment]
Thanks @eakmanrq! Congrats on your first commit 🎉 

So close to being the 3,000th contributor, but alas you are the 2,999th (still kinda cool imo).

**Comment by potiuk** [comment]
> So close to being the 3,000th contributor, but alas you are the 2,999th (still kinda cool imo).

👀👀👀👀👀👀👀👀👀👀👀

**Comment by eakmanrq** [comment]
> Thanks @eakmanrq! Congrats on your first commit 🎉
> 
> So close to being the 3,000th contributor, but alas you are the 2,999th (still kinda cool imo).

Wow congrats to you all! That says so much about the impact Airflow has had and the great community that has grown around it. 🚀
